### ISSUE TITLE
 lightweight engines  has one error about MissingPluginException(No implementation found for method incrementCount on channel multiple-flutters)

### ISSUE THREAD

**Issue by zewsummer** [flutter/flutter#97424]
## Steps to Reproduce

1. FlutterEngine is  2.8.1 Prepare FlutterFragment A and B, 
2. each fragmnet has an engine created with the flutterEngineGroup
2. each engine will registers all plug-ins channel,
2. however, an error will be reported every time when i used channel in fragment B。exception content： “Unhandled Exception: MissingPluginException(No implementation found for method xxxx)”
3. ... do not use flutterEngineGroup create engine,it's ok.
4. The official demo had the same problem，address：git@github.com:flutter/samples.git

Expected results:
we can use channel normally 

Actual results:
an error will be reported every time when i used channel in fragment A or B ，exception content： “Unhandled Exception: MissingPluginException(No implementation found for method xxxx)”

flutter doctor -v

`Flutter (Channel stable, 2.8.1, on Mac OS X 10.15.5 19F101 darwin-x64,
    locale zh-Hans-CN)
    • Flutter version 2.8.1 at /Users/jonyzhang/fvm/versions/2.8.1
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 77d935af4d (6 weeks ago), 2021-12-16 08:37:33 -0800
    • Engine revision 890a5fca2e
    • Dart version 2.15.1
    • Pub download mirror https://pub.flutter-io.cn

[!] Android toolchain - develop for Android devices (Android SDK version
    32.0.0)
    • Android SDK at /Users/jonyzhang/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for
      more details.

[!] Xcode - develop for iOS and macOS (Xcode 12.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.0.0.
      Download the latest version or update via the Mac App Store.
    ✗ CocoaPods not installed.
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin
        code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To install see
      https://guides.cocoapods.org/using/getting-started.html#installation for
      instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_202-release-1483-b49-5587405)`

**Comment by zewsummer** [comment]
engine version：2.8.1

**Comment by maheshmnj** [comment]
Hi @zewsummer, Thanks for filing the issue. If I understand correctly when you run the add to app sample (Android) and click on add you are seeing Missing Plugin exception is that correct?

if so, Have you tried running `flutter pub get` before running the application

```
cd ../multiple_flutters_module
flutter pub get
cd -
open -a "Android Studio" multiple_flutters_android/ # macOS command
# (build and run)
```

**Comment by zewsummer** [comment]
> open -a "Android Studio" multiple_flutters_android

yeah, thank you for reply,according to the procedure you said, there is still this exception，On the other hand, in my own project, the engine registration plug-in created by flutterEngineGroup often fails, please kindly check it

E/flutter ( 3065): [ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: MissingPluginException(No implementation found for method next on channel multiple-flutters)
E/flutter ( 3065): #0      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:165:7)
E/flutter ( 3065): <asynchronous suspension>
E/flutter ( 3065):

**Comment by maheshmnj** [comment]
@zewsummer I checked the multiple_flutters sample app and the app ran fine

> On the other hand, in my own project, the engine registration plug-in created by flutterEngineGroup often fails, please kindly check it

Can you please share a minimal and complete reproducible sample project  to investigate this further. 

https://user-images.githubusercontent.com/31410839/151748200-85fc6d6f-7d15-4567-9cbe-0f68a7263509.mov


>  ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for
      more details.


Additionally, Please try fixing these issues by downloading command line tools from Android studio under SDK tools
and accept the android license by running `flutter doctor --android-licenses`

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 2.8.1, on macOS 12.1 21C52 darwin-arm, locale en-GB)
    • Flutter version 2.8.1 at /Users/mahesh/Documents/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 77d935af4d (7 weeks ago), 2021-12-16 08:37:33 -0800
    • Engine revision 890a5fca2e
    • Dart version 2.15.1

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.10.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 60.1.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.63.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.32.0

[✓] Connected device (3 available)
    • Redmi K20 Pro (mobile) • 192.168.1.2:5555 • android-arm    • Android null (API null)
    • macOS (desktop)        • macos            • darwin-arm64   • macOS 12.1 21C52 darwin-arm
    • Chrome (web)           • chrome           • web-javascript • Google Chrome 97.0.4692.99

• No issues found!
```
</details>

**Comment by zewsummer** [comment]
This problem still exists, but I did not post it in the demo to reproduce, it is in multiple fragments instead of acitivity

**Comment by zewsummer** [comment]
![image](https://user-images.githubusercontent.com/31503964/153361764-1ef09004-a457-4c47-a960-210e56aaf288.png)

#

I've tried many times and I still get this problem，but he was an accident

**Comment by maheshmnj** [comment]
Thanks for the info I am unable to reproduce the issue the multiple flutters sample app builds fine on `stable 2.10`. 
Labeling this issue for further investigation.

**Comment by chinmaygarde** [comment]
[From here:](https://github.com/flutter/flutter/issues/97424#issuecomment-1034588930)
> multiple fragments instead of acitivity

Can we have the reproducible test case for this please? Reasoning about orchestrating plugin initialization using multiple engines is hard to do without this.

**Comment by zewsummer** [comment]
@chinmaygarde the official demo can be reproduced, but only occasionally。[git@github.com:flutter/samples.git](https://github.com/flutter/samples/tree/master/add_to_app/multiple_flutters)
